src/bin/common.rs

```diff
@@ -257,149 +258,175 @@ impl fmt::Display for FrameSummary {
   }
 }
 
+fn read_lookahead(ctx: &mut Context, y4m_dec: &mut y4m::Decoder<'_, Box<dyn Read>>, y4m_details: Y4MDetails) {
```

Comment:
Technically this function gets called even if we have 0 lookahead, right?

Reply:
This will get called but won't load any frames. I can add logic to the `needs_more_lookahead` method to handle that case. (Technically, the min lookahead should be 1.)

Reply:
Renamed to "read_frame_batch", since it may read one frame or it may read as many as it needs to fill the lookahead queue.